src/functional/ffront/past_passes/type_deduction.py

```diff
@@ -31,6 +31,11 @@ def _ensure_no_sliced_field(entry: past.Expr):
             _ensure_no_sliced_field(param)
 
 
+def _is_integral_scalar(scalar_type: past.Expr) -> bool:
```

Comment:
```suggestion
def _is_integral_scalar(expr: past.Expr) -> bool:
```
The argument is not necessarily a scalar and in no case a type.